main.cpp

```diff
@@ -131,6 +127,27 @@ struct FloatType
         FloatType& pow( const IntType& operand );
         FloatType& pow( float operand );
 
+        FloatType& apply(std::function<FloatType&(float&)> callable)
+        {
+            if(callable)
+            {
+                return callable(*heapFloat); 
+            }
+            else
+            {
+                return *this;
+            }
```

Comment:
You have no code that runs after the `else` block.  Therefore you do not need this `else` keyword.  You did it correctly in the `apply()` below.
- [ ] revise